packages/embark/src/lib/modules/ipfs/index.js

```diff
@@ -98,60 +128,74 @@ class IPFS {
   }
 
   addStorageProviderToEmbarkJS() {
-    this.events.request('version:downloadIfNeeded', 'ipfs-api', (err, location) => {
-      if (err) {
-        this.logger.error(__('Error downloading IPFS API'));
-        return this.logger.error(err.message || err);
-      }
-      this.events.request('code-generator:ready', () => {
-        this.events.request('code-generator:symlink:generate', location, 'ipfs-api', (err, _symlinkDest) => {
-          if (err) {
-            this.logger.error(__('Error creating a symlink to IPFS API'));
-            return this.logger.error(err.message || err);
-          }
-
-          this.events.emit('runcode:register', 'IpfsApi', require('ipfs-api'), () => {
-            let code = "";
-            code += "\nconst __embarkIPFS = require('embarkjs-ipfs')";
-            code += "\nEmbarkJS.Storage.registerProvider('ipfs', __embarkIPFS.default || __embarkIPFS);";
-
-            this.embark.addCodeToEmbarkJS(code);
-            this.embark.addConsoleProviderInit("storage", code, (storageConfig) => storageConfig.enabled);
+    if(!this.addedToEmbarkJs) {
+      this.addedToEmbarkJs = true;
+      this.events.request('version:downloadIfNeeded', 'ipfs-api', (err, location) => {
+        if (err) {
+          this.logger.error(__('Error downloading IPFS API'));
+          return this.logger.error(err.message || err);
+        }
+        this.events.request('code-generator:ready', () => {
+          this.events.request('code-generator:symlink:generate', location, 'ipfs-api', (err, _symlinkDest) => {
```

Comment:
I think this was added by Iuri, but just seeing it now. If the symLink is not used, we can remove the symlink creation

Reply:
Is the `ipfs-api` symlink not used?

Reply:
Doesn't look like it, because `_symlinkDest` is not used anywhere.
To be fair, I think the embarkjsipfs module is kind of broken is the fact that if you don,t use Embark's pipeline, the dapp will crash if you don't install the `embarkjsipfs` package.